src/appleseed/renderer/kernel/lighting/sppm/sppmlightingengine.cpp

```diff
@@ -292,11 +296,21 @@ namespace
 
                 // Emitted light.
                 if (vertex.m_edf && vertex.m_cos_on > 0.0)
-                    add_emitted_light_contribution(vertex, vertex_radiance);
+                {
+                    Spectrum emitted(0.0f, Spectrum::Illuminance);
+                    add_emitted_light_contribution(vertex, emitted);
```

Comment:
There's redundant work being done here. My advice: get rid of `add_emitted_light_contribution()` (inline it) and simplify things.